src/components/cylc/tree/InfiniteTree.vue

```diff
@@ -0,0 +1,208 @@
+import InfiniteTree from '../../../../node_modules/infinite-tree/dist/infinite-tree'
+
+const lcfirst = (str) => {
+  str += ''
+  return str.charAt(0).toLowerCase() + str.substr(1)
+}
+
+export default {
+  name: 'InfiniteTree',
+  components: {
+    RecycleScroller
+  },
+  props: {
+    treeData: {
+      type: [Array, Object],
+      default: () => {
+        return []
+      }
+    },
+    autoOpen: {
+      type: Boolean,
+      default: false
+    },
+    selectable: {
+      type: Boolean,
+      default: false
+    },
+    tabIndex: {
+      type: Number,
+      default: 1
+    },
+    rowHeight: {
+      type: Number,
+      default: 32
+    },
+    loadNodes: {
+      type: Function,
+      default: () => {}
+    },
+    shouldSelectNode: {
+      type: Function,
+      default: () => {}
+    },
+    shouldLoadNodes: {
+      type: Function,
+      default: () => {}
+    },
+    // Callback invoked before updating the tree.
+    onContentWillUpdate: {
+      type: Function,
+      default: null
+    },
+    // Callback invoked when the tree is updated.
+    onContentDidUpdate: {
+      type: Function,
+      default: null
+    },
+    // Callback invoked when a node is opened.
+    onOpenNode: {
+      type: Function,
+      default: null
+    },
+    // Callback invoked when a node is closed.
+    onCloseNode: {
+      type: Function,
+      default: null
+    },
+    // Callback invoked when a node is selected or deselected.
+    onSelectNode: {
+      type: Function,
+      default: null
+    },
+    // Callback invoked before opening a node.
+    onWillOpenNode: {
+      type: Function,
+      default: null
+    },
+    // Callback invoked before closing a node.
+    onWillCloseNode: {
+      type: Function,
+      default: null
+    },
+    // Callback invoked before selecting or deselecting a node.
+    onWillSelectNode: {
+      type: Function,
+      default: null
+    },
+    onKeyUp: {
+      type: Function,
+      default: null
+    },
+    onKeyDown: {
+      type: Function,
+      default: null
+    },
+    onMouseLeave: {
+      type: Function,
+      default: null
+    },
+    onMouseEnter: {
+      type: Function,
+      default: null
+    }
+  },
+  inheritAttrs: false,
+  data () {
+    return {
+      tree: {
+        nodes: []
+      },
+      nodes: [],
+      eventHandlers: {
+        onContentWillUpdate: null,
+        onContentDidUpdate: null,
+        onOpenNode: null,
+        onCloseNode: null,
+        onSelectNode: null,
+        onWillOpenNode: null,
+        onWillCloseNode: null,
+        onWillSelectNode: null,
+        onKeyUp: null,
+        onKeyDown: null,
+        onMouseEnter: null,
+        onMouseLeave: null
+      }
+    }
+  },
+  watch: {
+    treeData: {
+      handler (newValue) {
+        this.tree.loadData(newValue)
+      }
+    }
+  },
+  mounted () {
+    this.tree = new InfiniteTree({
+      el: this.$refs.tree,
+      ...this.$props,
```

Comment:
Note to self: this needs to be `this.options.$props`. `this.$props` doesn't seem to work on the `mounted` method (though I think it appears in docs but I think in other methods?)